test/Microsoft.AspNetCore.Sockets.Client.Tests/LongPollingTransportTests.cs

```diff
@@ -0,0 +1,167 @@
+                        ? HttpStatusCode.InternalServerError
+                        : HttpStatusCode.OK;
+                    return new HttpResponseMessage(statusCode) { Content = new StringContent(string.Empty) };
+                });
+
+            using (var httpClient = new HttpClient(mockHttpHandler.Object))
+            using (var longPollingTransport = new LongPollingTransport(httpClient, new LoggerFactory()))
+            {
+                var connectionToTransport = Channel.CreateUnbounded<Message>();
+                var transportToConnection = Channel.CreateUnbounded<Message>();
+                var channelConnection = new ChannelConnection<Message>(connectionToTransport, transportToConnection);
+                await longPollingTransport.StartAsync(new Uri("http://fakeuri.org"), channelConnection);
+
+                await connectionToTransport.Out.WriteAsync(new Message());
+
+                Assert.Equal(longPollingTransport.Running, await Task.WhenAny(Task.Delay(1000), longPollingTransport.Running));
+
+                // The channel needs to be drained for the Completion task to be completed
+                Message message;
+                while (transportToConnection.In.TryRead(out message))
+                {
+                    message.Dispose();
+                }
+
+                var exception = await Assert.ThrowsAsync<HttpRequestException>(async () => await transportToConnection.In.Completion);
+                Assert.Contains(" 500 ", exception.Message);
+            }
+        }
+
+        [Fact]
+        public async Task LongPollingTransportShutsDownWhenChannelIsClosed()
+        {
+            var mockHttpHandler = new Mock<HttpMessageHandler>();
+            mockHttpHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .Returns<HttpRequestMessage, CancellationToken>(async (request, cancellationToken) =>
+                {
+                    await Task.Yield();
+                    return new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent(string.Empty) };
+                });
+
+            using (var httpClient = new HttpClient(mockHttpHandler.Object))
+            using (var longPollingTransport = new LongPollingTransport(httpClient, new LoggerFactory()))
+            {
+                var connectionToTransport = Channel.CreateUnbounded<Message>();
+                var transportToConnection = Channel.CreateUnbounded<Message>();
+                var channelConnection = new ChannelConnection<Message>(connectionToTransport, transportToConnection);
+                await longPollingTransport.StartAsync(new Uri("http://fakeuri.org"), channelConnection);
+
+                connectionToTransport.Out.Complete();
+
+                Assert.Equal(longPollingTransport.Running, await Task.WhenAny(Task.Delay(1000), longPollingTransport.Running));
```

Comment:
I thought about this. I was thinking about creating a Testing.Common project that contains this kind of things since we would want to use it across different test projects in this repo.